jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/gzip/GzipResponseAndCallback.java

```diff
@@ -136,7 +136,13 @@ public void write(boolean last, ByteBuffer content, Callback callback)
             case NOT_COMPRESSING -> super.write(last, content, callback);
             case COMMITTING -> callback.failed(new WritePendingException());
             case COMPRESSING -> gzip(last, callback, content);
-            default -> callback.failed(new IllegalStateException("state=" + _state.get()));
+            default ->
+            {
+                if (BufferUtil.isEmpty(content))
+                    callback.succeeded();
```

Comment:
@lachlan-roberts This can happen is something does a flush when last has already been sent.    This can be needed when checking there is no data buffered in a writer.